### PR TITLE
API Server change TxOuput JSON format

### PR DESCRIPTION
**Pull request by OBorce**
- change Address and Amount outputs
- add Addressable for VRFPublicKey

### CODE REVIEW

File: common/src/address/hexified.rs [hunk 1]
```diff
@@ -337,47 +338,6 @@ mod tests {
         assert_eq!(res, "0xabcd");
     }
 
-    #[test]
-    fn invalid_match_should_replace_with_hex_creating_case_address_creation_error() {
```
Comment (TheQuantumPhysicist) [inline common/src/address/hexified.rs:341]: Like discussed in the daily, see if you can restore these two tests with other cases that test the same thing and arrive to the same result.

File: common/src/chain/config/mod.rs [hunk 2]
```diff
@@ -139,14 +139,14 @@ fn address_prefix(chain_type: ChainType, destination: &Destination) -> &'static
             Destination::ClassicMultisig(_) => "mmtc",
         },
         ChainType::Testnet => match destination {
-            Destination::AnyoneCanSpend => "",
+            Destination::AnyoneCanSpend => "anyonecanspend_address",
```
Comment (TheQuantumPhysicist) [inline common/src/chain/config/mod.rs:142]: Please change it for every chain type, and add a prefix signifying the net type.

Does it have to be so long, btw? Does something like `tanyonecanspend` not work?

File: common/src/chain/config/mod.rs [hunk 3]
```diff
@@ -245,6 +245,16 @@ impl ChainConfig {
...
     }
 
+    #[must_use]
+    pub fn vrf_public_key_address_prefix(&self) -> &'static str {
+        match self.chain_type {
+            ChainType::Mainnet => "mvrfpk",
+            ChainType::Testnet => "tvrfpk",
+            ChainType::Regtest => "rvrfpk",
+            ChainType::Signet => "svrfpk",
+        }
```
Comment (TheQuantumPhysicist) [inline common/src/chain/config/mod.rs:255]: Should the wallet use these now for displaying the VRF public key? I don't know if we do display it for any reason.

File: api-server/api-server-common/src/storage/impls/in_memory/transactional/read.rs [hunk 4]
```diff
@@ -48,6 +48,13 @@ impl<'t> ApiServerStorageRead for ApiServerInMemoryStorageTransactionalRo<'t> {
         self.transaction.get_block(block_id)
     }
 
+    async fn get_transaction_with_block(
```
Comment (TheQuantumPhysicist) [inline api-server/api-server-common/src/storage/impls/in_memory/transactional/read.rs:51]: Why do we need a function to return a transaction with a block?

**Review by TheQuantumPhysicist — APPROVED** [review]
Besides the comments, looks fine.

File: api-server/stack-test-suite/tests/v1/block_reward.rs [hunk 5]
```diff
@@ -169,7 +172,12 @@ async fn has_reward(#[case] seed: Seed) {
 
                     _ = tx.send((
                         block_index.block_id().to_hash().encode_hex::<String>(),
-                        json!(block.block_reward().outputs().iter().clone().collect::<Vec<_>>()),
+                        json!(block
+                            .block_reward()
+                            .outputs()
+                            .iter()
+                            .map(|out| txoutput_to_json(out, &chain_config))
+                            .collect::<Vec<_>>()),
```
Comment (iljakuklic) [inline api-server/stack-test-suite/tests/v1/block_reward.rs:180]: It's a test so no big deal by anyway: Is the `json!` macro needed here? I don't see any json-specific syntax, just a rust expression. Also `serde_json::Value` implements `from_iter` so you should be able  to `.collect::<serde_jsno::Value>()` if I'm not mistaken.

**Review by iljakuklic — APPROVED** [review]
Looks good to me

File: api-server/stack-test-suite/tests/v1/block_reward.rs (hunk 5 above)
Comment (OBorce) [reply api-server/stack-test-suite/tests/v1/block_reward.rs]: fixed with the collect to serde_json::Value

File: api-server/api-server-common/src/storage/impls/in_memory/transactional/read.rs (hunk 4 above)
Comment (OBorce) [reply api-server/api-server-common/src/storage/impls/in_memory/transactional/read.rs:51]: It is with the block aux data, used to get the block timestamp and height for the transaction endpoint's confirmations and timestamp.

File: common/src/chain/config/mod.rs (hunk 3 above)
Comment (OBorce) [reply common/src/chain/config/mod.rs:255]: changed the VRF public key printing in the wallet commands

File: common/src/chain/config/mod.rs (hunk 2 above)
Comment (OBorce) [reply common/src/chain/config/mod.rs]: used the shorted form works fine as well

**Pull request merged**